Feat/timer #13

071yoon opened the pull request:
## ✅ PR check list

<!--
하나씩 확인 후 체크박스에 표시해주세요.
-->

- [x] PR 제목: [feat/fix/refactor...] 작업 내용 한 줄 요약 (브랜치 이름) #이슈번호
- [x] commit message가 적절한지 확인해주세요.
- [x] 적절한 branch로 요청했는지 확인해주세요.
- [x] Assignees, Label을 붙여주세요.
- [x] 주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 Reviewer로 등록해주세요.
- [x] PR이 승인된 경우 해당 브랜치는 삭제해 주세요!

<br/>

## 📝 Changes

<!--
해당 pr에서 작업한 내역을 적어주세요.
처음엔 간단하게 요약, list 형식으로 세부사항 작성
-->

https://user-images.githubusercontent.com/66371206/154296387-280855c8-7d42-4f74-8107-9ce1ee08a6c5.mov

- Timer를 시각적으로 얼마나 남았는지 보여주는 컴포넌트입니다.
- gameTime 을 15, 30, 60 중 하나로 받아서 넘겨주면 각각 15초 30초 60초짜리 타이머로 보여줍니다
- 해당 영상은 30초짜리 타이머입니다

closes: #13

<!-- 해당 Issue Number 를 작성하세요!
무슨 이슈에 대한 작업인지 명시합니다. 해당 이슈를 이곳에 적으면 pr merge 이후 해당 이슈는 자동으로 close 됩니다. -->

<br/>

<!-- 테스트 사항이 있다면 -->

## 👨🏻‍💻 Test Checklist


## 📌 변경 및 주의 사항

+ 추후에 18번 줄인 console.log 를 찍는 부분이 서버에 시간이 끝났고, 자동제출을 하고 라운드가 넘어가게 되는 부분을 추가해야됩니다

+ 제가 처음에 테스트를 하다가 material/ui 를 넣어보다가 package-lock.json 에 그러한 부분들이 추가가 됐습니다. 그런데 로딩바로 쓰이기 딱 좋아서 일단 남겨뒀습니다. [로딩바로 쓰이기 좋은 Progress Bar](https://mui.com/components/progress/)

<!--
변경사항 및 주의 사항이 있다면 적어주세요.
주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 리뷰어로 등록해주세요. (다른 사람이 작성한 코드 수정 등)
코드 리뷰 시 더 꼼꼼하게 확인 받고 싶은 부분이 있다면 적어주세요.
-->

<br/>

071yoon committed:
:sparkles: 타이머 컴포넌트 생성

071yoon committed:
:sparkles: 타이머 컴포넌트 생성

071yoon committed:
pulling from develop for conflict
Merge branch 'develop' of https://github.com/AndroidNetrunner/decrypto into feat/Timer

AndroidNetrunner committed:
[ADD] rulebook

AndroidNetrunner reviewed (commented):
오오...! 저 원 디자인은 구현이 어려울 것 같아서 제가 포기한 건데...! 저걸 하시다니!

혹시 저 원 위에 남은 시간을 숫자로 쓰는 건 가능할까요?

071yoon commented:
> 



> 오오...! 저 원 디자인은 구현이 어려울 것 같아서 제가 포기한 건데...! 저걸 하시다니!
> 
> 혹시 저 원 위에 남은 시간을 숫자로 쓰는 건 가능할까요?

안그래도 제가 원 안에 남은 시간을 표현하려고 했는데, 디자인도 구리고 실력문제로 나아가지 못했슴미다... 추후에 제가 룰북 모달창 만들고 나서 다시 만져보겠습니다!!

071yoon committed:
pulling incase of collision
Merge branch 'develop' of https://github.com/AndroidNetrunner/decrypto into feat/Timer

071yoon committed:
Timer 컴포넌트에 시간 나오도록 수정 완료
시간에 걸맞는 폰트 추가 완료

071yoon committed:
Timer 컴포넌트에 시간 나오도록 수정 완료
시간에 걸맞는 폰트 추가 완료

071yoon commented:
https://user-images.githubusercontent.com/66371206/154440192-cc67bb61-d2c8-4836-912d-91237cb939dc.mov

수정 완료했습니다.

Seojunhwan commented on src/Components/Timer.tsx:
```diff
@@ -0,0 +1,64 @@
+  border-radius: 50%;
+  display: grid;
+  place-items: center;
+  .before {
```
`.before` 한 번 확인 부탁드립니당

071yoon replied:
넵 확인했습니다

AndroidNetrunner committed:
Merge branch 'develop' of https://github.com/AndroidNetrunner/decrypto into feat/rulebook

AndroidNetrunner committed:
[ADD] 제한시간이 끝날 경우 조치 설명

071yoon committed:
Merge pull request #31 from AndroidNetrunner/feat/rulebook

[DOCUMENT] rulebook.md

AndroidNetrunner reviewed (changes requested):
제가 원하던 그림이었습니다! 하지만 테스트를 위해...

Seojunhwan commented on src/Components/Timer.tsx:
```diff
@@ -0,0 +1,64 @@
+const Container = styled.div``;
+
+const TimerLeft = styled.div`
+  font-family: neodgm;
```
예비 폰트? 도 지정해주세요~!

071yoon replied:
넹

Seojunhwan reviewed (commented):
요거 하나만 확인해 주세요!

071yoon committed:
junseo님의 수정사항 반영

071yoon committed:
Merge branch 'feat/Timer' of https://github.com/AndroidNetrunner/decrypto into feat/Timer

Seojunhwan reviewed (approved):
음 코드 맛있네요~

AndroidNetrunner reviewed (approved):
👍
